libs/sdk-core/src/logging.rs

```diff
@@ -0,0 +1,112 @@
+use crate::{
+    error::{SdkError, SdkResult},
+    LogEntry, LogStream,
+};
+use anyhow::Result;
+use chrono::Local;
+use env_logger::Logger;
+use log::{Metadata, Record};
+use once_cell::sync::OnceCell;
+use std::io::Write;
+use std::{fs::OpenOptions, sync::RwLock};
+
+pub(crate) static SDK_LOGGER: OnceCell<RwLock<GlobalSdkLogger>> = OnceCell::new();
```

Comment:
IMO sooner or later we'll have to move away from a RW lock, because even getting a read lock includes reading / waiting / setting some internal locks:

- is write lock acquired now? then getting read lock has to wait for that
- once read lock is acquired, "write" the read lock state to prevent write access
- once read lock is relinquished, "write" the new read lock state

Its probably fine for now, but there are app usage scenarios where this can become a problem, for example frequent use of `set_log_directory`.